code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -50,6 +54,10 @@
 /obj/item/weapon/reagent_containers/food/drinks/afterattack(obj/target, mob/user , proximity)
 	if(!proximity) return
 	if(istype(target, /obj/structure/reagent_dispensers)) //A dispenser. Transfer FROM it TO us.
+	
+		if (!is_open_container())
```

Comment:
this is just all drinks and not soda's specifically, so now you have a soda specific message for all drinks